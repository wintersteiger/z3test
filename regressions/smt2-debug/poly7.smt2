
; Copyright (c) 2015 Microsoft Corporation

(set-logic QF_NRA) 
(declare-const x Real)
(declare-const y Real)

(poly/isolate-roots (+ x y 1) y 2)
(poly/isolate-roots (+ x y 1))
(poly/isolate-roots (+ x y 1) y (root-obj (+ (^ x 5) (* (- 1) x) (- 1)) 1))
(poly/isolate-roots (+ (* x x x) y 1) y (root-obj (+ (^ x 5) (* (- 1) x) (- 1)) 1))
(set-option :pp.decimal true)
(poly/isolate-roots (+ (* x x x) y 1) y (root-obj (+ (^ x 5) (* (- 1) x) (- 1)) 1))
(display (root-obj (+ (^ x 5) (* (- 1) x) (- 1)) 1))

(poly/isolate-roots (+ x y) y y)
(poly/isolate-roots (+ x y) 1 1)

(declare-const a Real)
(declare-const b Real)

(display (root-obj (+ (* 2 (^ x 2)) (* 3 x) (- 3)) 1))
(display (root-obj (- (^ x 2) 2) 1))

