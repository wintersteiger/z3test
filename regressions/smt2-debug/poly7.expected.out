(roots
(root-obj (+ x 3) 1))
(error "line 9 column 29: given assignment is not sufficient to make the given polynomial univariate")
(roots
(root-obj (+ (^ x 5) (* 5 (^ x 4)) (* 10 (^ x 3)) (* 10 (^ x 2)) (* 4 x) 1) 1))
(roots
(root-obj (+ (^ x 15) (* 5 (^ x 12)) (* 10 (^ x 9)) (* 10 (^ x 6)) (* 4 (^ x 3)) 1) 1))
(roots
-1.2941162066?)
1.1673039782?
(error "line 16 column 31: invalid assignment, argument is not a value")
(error "line 17 column 30: invalid assignment, argument is not a variable in the given polynomial")
(- 2.1861406616?)
(- 1.4142135623?)
